### PR TITLE
Warn when an incompatible NuGet and MSBuild are used together

### PR DESCRIPTION
**Pull request by johnterickson**

### CODE REVIEW

File: Tasks/NuGetCommandV2/nugetcommandmain.ts [hunk 1]
```diff
@@ -30,6 +31,18 @@ async function main(): Promise<void> {
...
+
+        const nugetSemVer = semver.coerce(nugetVersion);
+        if (semver.lt(nugetSemVer, '4.8.0'))
+        {
+            const msbuildSemVer = await nuGetGetter.getMSBuildVersion();
+            if (semver.gte(msbuildSemVer, '16.5.0'))
+            {
+                tl.logIssue(
+                    tl.IssueType.Warning, 
+                    "The detected version of MSBuild (" + msbuildSemVer + ") is incompatible with NuGet (" + nugetSemVer + "). See https://aka.ms/AA84eyr");
```
Comment (satbai) [inline Tasks/NuGetCommandV2/nugetcommandmain.ts:43]: ❤️

File: Tasks/NuGetCommandV2/nugetcommandmain.ts [hunk 2]
```diff
@@ -30,6 +31,18 @@ async function main(): Promise<void> {
         if (nugetVersionInfo && nugetVersionInfo.fileVersion){
             nugetVersion = nugetVersionInfo.fileVersion.toString();
         }
+
+        const nugetSemVer = semver.coerce(nugetVersion);
+        if (semver.lt(nugetSemVer, '4.8.0'))
```
Comment (satbai) [inline Tasks/NuGetCommandV2/nugetcommandmain.ts:36]: I may be mistaken but I thought it was 4.8.2?

File: Tasks/NuGetCommandV2/nugetcommandmain.ts [hunk 3]
```diff
@@ -30,6 +31,18 @@ async function main(): Promise<void> {
...
+        {
+            const msbuildSemVer = await nuGetGetter.getMSBuildVersion();
+            if (semver.gte(msbuildSemVer, '16.5.0'))
```
Comment (satbai) [inline Tasks/NuGetCommandV2/nugetcommandmain.ts:39]: We already call getNuGet() earlier in this file and getNuGet() also has this warning so I think it's sufficient to only have it there.

Comment (johnterickson) [reply Tasks/NuGetCommandV2/nugetcommandmain.ts:39]: Yeah the earlier call sometimes short circuits before we get to the code path we need.

**Comment by Automobileservices4** [comment]
#12652

**Pull request merged**